### PR TITLE
#1792-Enable application details for public institution user

### PR DESCRIPTION
**Pull request by guru-aot**
Enable application details page for public institution user

**Acceptance Criteria**
- [x] Enable authorization paths to Application details for Public institution user.
- [x] Enable application to be visualized for Public institution user.
- [x] Build E2E test.

The sidebar for the application details is added till students, @ann-aot  will update the assessments part.
image

Additional screenshots on the Application details page.
image

### CODE REVIEW

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+          applicationNumber: savedApplication.applicationNumber,
+          applicationFormName: "SFAA2022-23",
+          applicationProgramYearID: savedApplication.programYearId,
+        });
+    },
+  );
+
+  it(
+    "Should not have access to get the student application details when student has a submitted application " +
+      "for a different institution (application with location id saved).",
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts:91]: what do you mean by `for a different institution` here, did u mean a non-public institution?

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        .expect(HttpStatus.FORBIDDEN)
+        .expect({
+          statusCode: 403,
+          message: "Forbidden resource",
+          error: "Forbidden",
+        });
+    },
+  );
+
+  it("Should not get the student application details when application is submitted for different institution ", async () => {
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts:120]: 89 and 120 test description looks similar

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    // Arrange
+    // Student has a submitted application to the institution.
+    const student = await saveFakeStudent(appDataSource);
+    //Create new application.
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts:124]: can add a space before `Create`. line 96, 61 too

File: sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    private readonly applicationService: ApplicationService,
+    private readonly applicationControllerService: ApplicationControllerService,
+  ) {
+    super();
+  }
+
+  /**
+   * API to fetch application details by applicationId.
+   * This API will be used by ministry users.
+   * @param applicationId
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts:32]: can u add brief descriptions to the params

File: sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  constructor(
+    private readonly applicationService: ApplicationService,
+    private readonly applicationControllerService: ApplicationControllerService,
+  ) {
+    super();
+  }
+
+  /**
+   * API to fetch application details by applicationId.
+   * This API will be used by ministry users.
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts:31]: what does this mean line 31?

File: sources/packages/web/src/router/AESTRoutes.ts [hunk 6]
```diff
@@ -19,7 +19,7 @@ import Designation from "@/views/aest/institution/Designation.vue";
 import Restrictions from "@/views/aest/institution/Restrictions.vue";
 import FileUploads from "@/views/aest/student/FileUploads.vue";
 import InstitutionNotes from "@/views/aest/institution/InstitutionNotes.vue";
-import ApplicationDetails from "@/views/aest/ApplicationDetails.vue";
+import ApplicationDetails from "@/components/common/ApplicationDetails.vue";
```
Comment (ann-aot) [inline sources/packages/web/src/router/AESTRoutes.ts:22]: why this change?? views should from `views/` folder

File: sources/packages/web/src/router/InstitutionRoutes.ts [hunk 7]
```diff
@@ -5,6 +5,7 @@ import InstitutionCreate from "@/views/institution/InstitutionCreate.vue";
 import InstitutionUserProfile from "@/views/institution/InstitutionUserProfile.vue";
 import AppInstitution from "@/views/institution/AppInstitution.vue";
 import ManageLocation from "@/views/institution/ManageLocations.vue";
+import ApplicationDetails from "@/components/common/ApplicationDetails.vue";
```
Comment (ann-aot) [inline sources/packages/web/src/router/InstitutionRoutes.ts:8]: views should from `views/` folder

File: sources/packages/web/src/router/InstitutionRoutes.ts [hunk 8]
```diff
@@ -25,6 +26,7 @@ import { ClientIdType } from "@/types/contracts/ConfigContract";
 import { AuthStatus, AppRoutes, InstitutionUserTypes } from "@/types";
 import ManageInstitutionSideBar from "@/components/layouts/Institution/sidebar/ManageInstitutionSideBar.vue";
 import InstitutionHomeSideBar from "@/components/layouts/Institution/sidebar/HomeSideBar.vue";
+import InstitutionApplicationSideBar from "@/components/layouts/Institution/sidebar/InstitutionApplicationSideBar.vue";
```
Comment (ann-aot) [inline sources/packages/web/src/router/InstitutionRoutes.ts:29]: views should from `views/` folder

File: sources/packages/web/src/router/InstitutionRoutes.ts [hunk 9]
```diff
@@ -544,6 +547,36 @@ export const institutionRoutes: Array<RouteRecordRaw> = [
...
+            InstitutionUserTypes.user,
+          ],
+        },
+        children: [
+          {
+            path: "",
+            name: InstitutionRoutesConst.STUDENT_APPLICATION_DETAILS,
+            props: true,
+            component: InstitutionApplicationView,
+            meta: {
```
Comment (ann-aot) [inline sources/packages/web/src/router/InstitutionRoutes.ts:570]: I believe we don't need these meta for children if the parent has the same meta

File: sources/packages/web/src/services/http/ApplicationApi.ts [hunk 10]
```diff
@@ -79,15 +79,18 @@ export class ApplicationApi extends HttpBaseClient {
 
   /**
    * API Client for application detail.
-   * @param applicationId
-   * @returns
+   * @param applicationId for the application.
```
Comment (ann-aot) [inline sources/packages/web/src/services/http/ApplicationApi.ts:82]: 👍

File: sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue [hunk 11]
```diff
@@ -0,0 +1,78 @@
+<template>
+  <full-page-container v-if="applicationDetail.data" class="my-2">
```
Comment (ann-aot) [inline sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue:2]: do we still need this class

File: sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue [hunk 12]
```diff
@@ -0,0 +1,78 @@
+<template>
```
Comment (ann-aot) [inline sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue:1]: just wondering, do we need to make this page a component? dev thoughts?

**Review by ann-aot — COMMENTED** [review]
Good work @guru-aot . Added some comments.

File: sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+   * @param applicationId
+   * @param studentId
+   * @returns Application details.
+   */
+  @HasStudentDataAccess("studentId")
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts:36]: Trying to follow what we did for the locations, I believe that the student should be at the root of the endpoint, for instance, similar to what we did for the below one.
https://github.com/bcgov/SIMS/blob/main/sources/packages/backend/apps/api/src/route-controllers/program-info-request/program-info-request.institutions.controller.ts#L84

To get more context, search the source code by "@HasLocationAccess("locationId")" and check how we did it.

For this case, for instance, it would be

```
api/institutions/institution/student/:studentId/application/:application
```

@ann-aot @andrepestana-aot @sh16011993 @dheepak-aot thoughts?

File: sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+        loadDynamicData: true,
+        studentId: studentId,
+        institutionId: userToken.authorizations.institutionId,
+      },
+    );
+    application.data =
+      await this.applicationControllerService.generateApplicationFormData(
+        application.data,
+      );
+    return this.applicationControllerService.transformToApplicationForDTO(
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts:55]: O noticed that the method `transformToApplicationForDTO` is centralized but instead of having it centralized can we have the combination of the `generateApplicationFormData` + `transformToApplicationForDTO`?

File: sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+  AllowAuthorizedParty,
+  HasStudentDataAccess,
+  IsBCPublicInstitution,
+  UserToken,
+} from "../../auth/decorators";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { ApiTags } from "@nestjs/swagger";
+import { ClientTypeBaseRoute } from "../../types";
+import { ApplicationControllerService } from "./application.controller.service";
+import { IInstitutionUserToken } from "../../auth";
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts:15]: I believe that it can be combined as below
```ts
import { AuthorizedParties, IInstitutionUserToken } from "../../auth";
```

File: sources/packages/backend/apps/api/src/services/application/application.service.ts [hunk 16]
```diff
@@ -553,6 +558,9 @@ export class ApplicationService extends RecordDataModelService<Application> {
         location: {
           id: true,
           name: true,
+          institution: {
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/services/application/application.service.ts:561]: Are we adding this extra select and the `location` on `relations` because of the where condition or is it needed to be returned in the method now?

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts [hunk 17]
```diff
@@ -0,0 +1,150 @@
...
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCLocation,
+    );
+  });
+
+  it(
+    "Should get the student application details when student has a submitted application " +
+      "for the institution (application with location id saved).",
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts:56]: I gave the same comment on the previous PR following the same, I would prefer not to have the explanation "(application with location id saved)".

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts [hunk 18]
```diff
@@ -0,0 +1,150 @@
...
+    async () => {
+      // Arrange
+      // Student has a submitted application to the institution.
+      const student = await saveFakeStudent(appDataSource);
+      //Create new application.
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts:61]: Please add a space after "//".

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts [hunk 19]
```diff
@@ -0,0 +1,150 @@
...
+    );
+  });
+
+  it(
+    "Should get the student application details when student has a submitted application " +
+      "for the institution (application with location id saved).",
+    async () => {
+      // Arrange
+      // Student has a submitted application to the institution.
+      const student = await saveFakeStudent(appDataSource);
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts:60]: Is there a benefit on creating the student here instead of allowing the `saveFakeStudent` do it?

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts [hunk 20]
```diff
@@ -0,0 +1,150 @@
...
+  );
+
+  it(
+    "Should not have access to get the student application details when student has a submitted application " +
+      "for a different institution (application with location id saved).",
+    async () => {
+      // Arrange
+      // Student has a submitted application to the institution.
+      const student = await saveFakeStudent(appDataSource);
+      //Create new application.
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts:96]: Same from https://github.com/bcgov/SIMS/pull/1914/files#r1183173489

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts [hunk 21]
```diff
@@ -0,0 +1,150 @@
...
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.FORBIDDEN)
+      .expect({
+        statusCode: 403,
+        message: "Forbidden resource",
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts:142]: I made a comment on @dheepak-aot PR that can change this check https://github.com/bcgov/SIMS/pull/1915#discussion_r1183142941

File: sources/packages/web/src/router/InstitutionRoutes.ts [hunk 22]
```diff
@@ -544,6 +547,36 @@ export const institutionRoutes: Array<RouteRecordRaw> = [
           },
         ],
       },
+      {
+        path: AppRoutes.ApplicationDetail,
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/router/InstitutionRoutes.ts:551]: Are we missing the `allowOnlyBCPublic` here?

File: sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue [hunk 23]
```diff
@@ -0,0 +1,78 @@
...
+        }"
+        subTitle="Financial Aid Application"
+      >
+      </header-navigator>
+    </template>
+    <h2 class="color-blue pb-4">
+      Student Application Details
+      {{
+        applicationDetail.applicationNumber
+          ? " - " + applicationDetail.applicationNumber
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue:18]: Please use the formatter.

File: sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue (hunk 12 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue:1]: I see the point @ann-aot but I am not sure since the main part of this page (IMO) is the `StudentApplication` that is already a component.

**Review by andrewsignori-aot — COMMENTED** [review]
Great work, please take a look at the comments.

File: sources/packages/backend/apps/api/src/route-controllers/application/application.aest.controller.ts [hunk 24]
```diff
@@ -52,7 +52,7 @@ export class ApplicationAESTController extends BaseController {
       await this.applicationControllerService.generateApplicationFormData(
         application.data,
       );
-    return this.applicationControllerService.transformToApplicationForAESTDTO(
+    return this.applicationControllerService.transformToApplicationForDTO(
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/application.aest.controller.ts:55]: Suggestion: tansformToApplicationDTO

File: sources/packages/backend/apps/api/src/route-controllers/application/application.controller.service.ts [hunk 25]
```diff
@@ -127,7 +127,7 @@ export class ApplicationControllerService {
    * @param application
    * @returns Application DTO
    */
-  async transformToApplicationForAESTDTO(
+  async transformToApplicationForDTO(
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/application.controller.service.ts:130]: Same suggestion.

File: sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts (hunk 5 above)
Comment (dheepak-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts:31]: ministry users?

File: sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts (hunk 13 above)
Comment (dheepak-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts:36]: I agree on the basis that it puts the context more clearly and I believe in that case, this end point must move to student.institutions.controller right?

File: sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue (hunk 12 above)
Comment (ann-aot) [reply sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue:1]: ya, I was also having a double thought

File: sources/packages/backend/apps/api/src/services/application/application.service.ts (hunk 16 above)
Comment (guru-aot) [reply sources/packages/backend/apps/api/src/services/application/application.service.ts:561]: returned as part of where condition

Comment (dheepak-aot) [reply sources/packages/backend/apps/api/src/services/application/application.service.ts:561]: @guru-aot if that is the case, I believe we can remove the `institution:true` in the relations and leave it as it is. Because the where will work without relevance in relations.

File: sources/packages/backend/apps/api/src/services/application/application.service.ts [hunk 26]
```diff
@@ -513,16 +513,21 @@ export class ApplicationService extends RecordDataModelService<Application> {
...
+   * - `institutionId` institution id.
    * @returns student application.
    */
   async getApplicationById(
     applicationId: number,
-    options?: { loadDynamicData?: boolean; studentId?: number },
+    options?: {
+      loadDynamicData?: boolean;
+      studentId?: number;
+      institutionId?: number;
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/api/src/services/application/application.service.ts:529]: @guru-aot  @andrewsignori-aot @ann-aot @andrepestana-aot  @sh16011993 

I was wondering that, with so many options in place which justifies the service as such but in this particular scenario where this service is required to be used with studentId and institutionId. Hence from the consumption perspective we need application details of a student who applied to a given institution and IMO we can introduce a service which is more confined to do this by re-using existing service. 

like this

```ts
async getApplicationByStudentAndInstitution(
    applicationId: number,
    studentId: number,
    institutionId: number,
    options?: { loadDynamicData?: boolean },
  ) {
    return this.getApplicationById(applicationId, {
      studentId,
      institutionId,
      loadDynamicData: options?.loadDynamicData,
    });
  }
```

File: sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts (hunk 13 above)
Comment (ann-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts:35]: did you mean `api/institutions/student/:studentId/application/:application` ?, if it s `api/institutions/institution/student/:studentId/application/:application` it will be `institution.institutions.controller` right?

File: sources/packages/web/src/router/InstitutionRoutes.ts [hunk 27]
```diff
@@ -544,6 +547,36 @@ export const institutionRoutes: Array<RouteRecordRaw> = [
...
+        meta: {
+          clientType: ClientIdType.Institution,
+          institutionUserTypes: [
+            InstitutionUserTypes.admin,
+            InstitutionUserTypes.user,
+          ],
+        },
+        children: [
+          {
+            path: "",
```
Comment (dheepak-aot) [inline sources/packages/web/src/router/InstitutionRoutes.ts:566]: Path empty?

File: sources/packages/web/src/router/InstitutionRoutes.ts (hunk 22 above)
Comment (dheepak-aot) [reply sources/packages/web/src/router/InstitutionRoutes.ts:551]: Yes is is missing and required here.

File: sources/packages/backend/apps/api/src/services/application/application.service.ts (hunk 26 above)
Comment (ann-aot) [reply sources/packages/backend/apps/api/src/services/application/application.service.ts:529]: what is the benefit here @dheepak-aot? I think I am not getting it

File: sources/packages/web/src/router/InstitutionRoutes.ts (hunk 27 above)
[the remark this answers is not in the document]
Comment (ann-aot) [reply sources/packages/web/src/router/InstitutionRoutes.ts:566]: I think the when even when the parent URL is entered, this will ensure that a child component is rendered. If we need to give a value here, then we will need to add logic to redirect to the child when the parent URL is requested. else we will get a blank or a partial page

File: sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts (hunk 14 above)
Comment (guru-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts]: The implementation is extended to students too with an implementation to get the first coe in between these 2 helpers.

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts (hunk 19 above)
Comment (guru-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts:62]: Having the student creation in the beforeall, fails the last test case. So to ignore the conflict, creating it seperately.

File: sources/packages/web/src/views/aest/StudentApplicationView.vue [hunk 28]
```diff
@@ -15,7 +15,7 @@
       Student Application Details
       {{
         applicationDetail.applicationNumber
-          ? " - " + applicationDetail.applicationNumber
+          ? HYPHEN_WITH_SPACE + applicationDetail.applicationNumber
```
Comment (dheepak-aot) [inline sources/packages/web/src/views/aest/StudentApplicationView.vue:18]: Will there be an application here without number? We can skip the expression if there is no such possibility.

File: sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue [hunk 29]
```diff
@@ -0,0 +1,80 @@
...
+        }"
+        subTitle="Financial Aid Application"
+      >
+      </header-navigator>
+    </template>
+    <h2 class="color-blue pb-4">
+      Student Application Details
+      {{
+        applicationDetail.applicationNumber
+          ? HYPHEN_WITH_SPACE + applicationDetail.applicationNumber
```
Comment (dheepak-aot) [inline sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue:18]: https://github.com/bcgov/SIMS/pull/1914/files#r1184384600

File: sources/packages/web/src/router/InstitutionRoutes.ts (hunk 27 above)
[the remark this answers is not in the document]
Comment (ann-aot) [reply sources/packages/web/src/router/InstitutionRoutes.ts:567]: yup, I also noticed that, and it is a gap, IMO. 

`or put a value to path and assume the user is not suppose to use the parent URL as it is not complete.`, are we suppose to assume this? 😉

[the remark this answers is not in the document]
Comment (ann-aot) [reply sources/packages/web/src/router/InstitutionRoutes.ts:567]: _I think, when the parent URL is entered, this will ensure that a child component is rendered (a complete view/page). If we want to add a value here, then we will need to add logic to redirect to the child when the parent URL is requested. else we will get a blank or a partial page/view_


As I mentioned above there are 2 ways, 
1. one in this example
2. add redirect logic, something like `path: 'child-path', redirect: 'parent -path' ` this https://router.vuejs.org/guide/essentials/redirect-and-alias.html,

[the remark this answers is not in the document]
Comment (ann-aot) [reply sources/packages/web/src/router/InstitutionRoutes.ts:567]: > @ann-aot by mentioning `when parent URL is entered` I believe you meant by manually typing the URL in browser right?
> 
> If that is the case, then we have the same situtation right now in our application where if we type a parent URL instead of child in AEST/Institution student search result i.e student details parent URL, it appears to be like this.
> 
> image
> 
> So we can either follow this way of having `path=""` for student profile route which is default for the parent or put a value to path and assume the user is not suppose to use the parent URL as it is not complete.


As I mentioned above there are 2 ways, 
1. one in this example
2. add redirect logic, something like `path: 'child-path', redirect: 'parent -path' ` this https://router.vuejs.org/guide/essentials/redirect-and-alias.html,

[the remark this answers is not in the document]
Comment (ann-aot) [reply sources/packages/web/src/router/InstitutionRoutes.ts:567]: > I think, when the parent URL is entered, this will ensure that a child component is rendered (a complete view/page). If we want to add a value here, then we will need to add logic to redirect to the child when the parent URL is requested. else we will get a blank or a partial page/view


As I mentioned above there are 2 ways, 
1. one in this example
2. add redirect logic, something like `path: 'child-path', redirect: 'parent -path' ` this https://router.vuejs.org/guide/essentials/redirect-and-alias.html, 


REF: https://router.vuejs.org/guide/essentials/nested-routes.html

File: sources/packages/web/src/router/InstitutionRoutes.ts (hunk 9 above)
Comment (ann-aot) [reply sources/packages/web/src/router/InstitutionRoutes.ts]: did u test this scenario @guru-aot . just to confirm

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts (hunk 17 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts]: Thanks for removing the `(application with location id saved)`.
Do we really need to break the line to have the `"for the institution."` in the second part?

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts (hunk 19 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts:62]: I am not sure how we end up having the `beforeall` in this conversation. The only need that I recommended was leaving the student creation to the `saveFakeApplication`.
Please check if the below code makes sense or maybe I am missing something.
```ts
       // Create new application.
      const savedApplication = await saveFakeApplication(appDataSource, {
        institution: collegeF,
        institutionLocation: collegeFLocation,
      });
      const student = savedApplication.student;

      const endpoint = `/institutions/application/${savedApplication.id}/student/${student.id}`;
      const institutionUserToken = await getInstitutionToken(
        InstitutionTokenTypes.CollegeFUser,
      );
```

File: sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue (hunk 23 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue]: Please use the `emptyStringFiller` from the `useFormatters()`.

File: sources/packages/backend/apps/api/src/services/application/application.service.ts (hunk 26 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/services/application/application.service.ts:529]: I am not sure if I am following. Right now the method is consumed 3 times where the only parameter that is required is the `applicationId`. The idea would be to have more methods where the parameters would be required?

File: sources/packages/web/src/composables/useFormatters.ts [hunk 30]
```diff
@@ -10,6 +10,7 @@ dayjs.extend(customParseFormat);
 
 const DEFAULT_EMPTY_VALUE = "-";
 export const DATE_ONLY_ISO_FORMAT = "YYYY-MM-DD";
+export const HYPHEN_WITH_SPACE = " - ";
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/composables/useFormatters.ts:13]: Please see https://github.com/bcgov/SIMS/pull/1914#discussion_r1184417241

File: sources/packages/web/src/views/aest/StudentApplicationView.vue (hunk 28 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/views/aest/StudentApplicationView.vue:18]: https://github.com/bcgov/SIMS/pull/1914#discussion_r1184417241

File: sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue (hunk 29 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue:18]: https://github.com/bcgov/SIMS/pull/1914#discussion_r1184417241

File: sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts (hunk 13 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts:35]: @ann-aot @dheepak-aot trying to add more information.
Currently, we have already implemented the below endpoints:
- api/institutions/student/251
- api/institutions/student/251/application-summary
- api/institutions/student/251/documents
- api/institutions/overaward/student/251/balance (this one is already not following the others)

The suggested one would be as below (as Ann mentioned)
- api/institutions/student/251/application/:application

File: sources/packages/web/src/composables/useFormatters.ts (hunk 30 above)
Comment (guru-aot) [reply sources/packages/web/src/composables/useFormatters.ts]: removed it

File: sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue (hunk 29 above)
Comment (guru-aot) [reply sources/packages/web/src/views/institution/student/InstitutionStudentApplicationView.vue]: removed it and used emptyStringFiller

File: sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts (hunk 13 above)
Comment (guru-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts:35]: As discussed, changed the controller

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts [hunk 31]
```diff
@@ -0,0 +1,136 @@
...
+        data: {},
+        id: savedApplication.id,
+        applicationStatus: savedApplication.applicationStatus,
+        applicationNumber: savedApplication.applicationNumber,
+        applicationFormName: "SFAA2022-23",
+        applicationProgramYearID: savedApplication.programYearId,
+      });
+  });
+
+  it("Should not have access to get the student application details if the student submitted and application to non-public institution.", async () => {
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts:83]: Minor: we're using "Should/when" in the test cases.

**Review by ann-aot — APPROVED** [review]
LGTM @guru-aot 👍. Thanks for doing the changes

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the changes, looks good 👍

**Review by andrepestana-aot — APPROVED** [review]
Looks good!

File: sources/packages/backend/apps/api/src/services/application/application.service.ts (hunk 26 above)
Comment (dheepak-aot) [reply sources/packages/backend/apps/api/src/services/application/application.service.ts:529]: That is right @andrewsignori-aot. Not part of this PR. But we can discuss outside about this idea.
@ann-aot It was exactly what @andrewsignori-aot below your comment.

File: sources/packages/web/src/router/InstitutionRoutes.ts (hunk 27 above)
[the remark this answers is not in the document]
Comment (dheepak-aot) [reply sources/packages/web/src/router/InstitutionRoutes.ts]: Yes, something we can discuss outside the PR.

**Review by dheepak-aot — APPROVED** [review]
Thanks for doing the changes @guru-aot 👍LGTM.
